src/main/kotlin/org/jitsi/nlj/rtp/ParsedVideoPacket.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.rtp
+
+/**
+ * A [VideoRtpPacket] which has been parsed and therefore
+ * contains more information, such as whether or not it's
+ * a keyframe
+ */
+abstract class ParsedVideoPacket(
+    buffer: ByteArray,
+    offset: Int,
+    length: Int
+) : VideoRtpPacket(buffer, offset, length) {
+
+    abstract val isKeyframe: Boolean
+    abstract val spatialLayerIndex: Int
+    var qualityIndex: Int = -1
```

Comment:
Are `spatialLayerIndex` and `qualityIndex` actually different?

Reply:
my understanding is that they are.  @gpolitis is the architect here, but my understanding is that qualiyIndex is an overall index across all the encodings (i.e. would be from 0-8 for 3 spatial layers, each with 3 temporal layers)

Reply:
Is `qualityIndex` currently getting set anywhere?

Reply:
It looks like it's getting set in `VideoParser.transform`, but it doesn't seem to be being _used_ anywhere.

Reply:
it's used in a couple places in `AdaptiveTrackProjection

Reply:
Oh, yes, I see, IntelliJ didn't find that usage of it.

Isn't that code going to break with your change, since `VideoRtpPacket.qualityIndex` doesn't exist anymore?

Reply:
yeah, i think i wrote it in the original message i sent you, should've added it here, but there are some changes required on the bridge.  basically just changes the type used from `VideoRtpPacket` to `ParsedVideoPacket`